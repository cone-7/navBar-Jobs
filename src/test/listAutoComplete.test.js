import React from 'react';
import ListAutoComplete from '../components/listAutoComplete.js';
import renderer from 'react-test-renderer';

test('Render ListAutoComplete', () => {
  const list = [
    {normalized_job_title:'test',uuid:'123'},
    {normalized_job_title:'test2',uuid:'1234'}
  ]
  const component = renderer.create(
    <ListAutoComplete list={list}></ListAutoComplete>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('clicked element of list', () => {
    const list = [
        {normalized_job_title:'test',uuid:'123'},
        {normalized_job_title:'test2',uuid:'1234'}
      ]
    const component = renderer.create(
    <ListAutoComplete list={list} clicked={() => console.log(1)}></ListAutoComplete>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree.children[0].props.onClick();
    // re-rendering
    tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  
});
