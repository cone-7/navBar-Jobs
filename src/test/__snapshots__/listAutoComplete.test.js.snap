// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ListAutoComplete 1`] = `
<div
  className="list-Auto-Complete"
>
  <span
    onClick={[Function]}
  >
    test
  </span>
  <span
    onClick={[Function]}
  >
    test2
  </span>
</div>
`;

exports[`clicked element of list 1`] = `
<div
  className="list-Auto-Complete"
>
  <span
    onClick={[Function]}
  >
    test
  </span>
  <span
    onClick={[Function]}
  >
    test2
  </span>
</div>
`;

exports[`clicked element of list 2`] = `
<div
  className="list-Auto-Complete"
>
  <span
    onClick={[Function]}
  >
    test
  </span>
  <span
    onClick={[Function]}
  >
    test2
  </span>
</div>
`;
